test/modules/DirectMethodSender/Settings.cs

```diff
@@ -106,13 +110,15 @@ public override string ToString()
                 { nameof(this.TestStartDelay), this.TestStartDelay.ToString() },
                 { nameof(this.TrackingId), this.TrackingId.ToString() },
                 { nameof(this.TransportType), Enum.GetName(typeof(TransportType), this.TransportType) },
+                { nameof(this.DirectMethodName), this.DirectMethodName },
                 { nameof(this.DirectMethodDelay), this.DirectMethodDelay.ToString() },
                 { nameof(this.InvocationSource), this.InvocationSource.ToString() },
+                { nameof(this.DirectMethodResultType), this.DirectMethodResultType.ToString() },
             };
 
-            this.AnalyzerUrl.ForEach((url) =>
+            this.ReportingEndpointUrl.ForEach((url) =>
```

Comment:
Remove this ForEach; and call "fields.Add(nameof(this.ReportingEndpointUrl), this.ReportingEndpointUrl.ToString());"